src/box/replication.cc

```diff
@@ -517,7 +552,6 @@ replica_on_applier_sync(struct replica *replica)
 {
 	if (replica->applier_sync_state == APPLIER_SYNC)
 		return;
-	assert(replica->applier_sync_state == APPLIER_CONNECTED);
```

Comment:
Instead of removing the assertion, better extend it with all possible conditions and add an explanatory comment.

Reply:
Discussed f2f and decided to leave as is because shutdown can happen at any applier state.